File: src/main/java/no/nav/pensjon/regler/domain/grunnlag/Garantipensjonsbeholdning.java
package no.nav.pensjon.regler.domain.grunnlag;

import no.nav.pensjon.regler.domain.beregning2011.JustertGarantipensjonsniva;
import no.nav.pensjon.regler.domain.kode.BeholdningsTypeCti;
import no.nav.pensjon.regler.domain.kode.GarantipenNivaCti;

public class Garantipensjonsbeholdning extends Beholdning {
    private static final long serialVersionUID = 7094272502601985660L;

    private JustertGarantipensjonsniva justertGarantipensjonsniva;
    private double pensjonsbeholdning;
    private double delingstall67;

    /**
     * Satstype brukt i garantipensjonsnivå.
     */
    private GarantipenNivaCti satsType;

    /**
     * Garantipensjonsnivå sats
     */
    private double sats;

    /**
     * Garantipensjonsnivå justert for trygdetid
     */
    private double garPN_tt_anv;

    /**
     * Garantipensjonsnivå fremskrevet.
     */
    private double garPN_justert;

    public Garantipensjonsbeholdning() {
        super();
        beholdningsType = new BeholdningsTypeCti("GAR_PEN_B");
    }
    public Garantipensjonsbeholdning(Garantipensjonsbeholdning garb) {
        super(garb);
        pensjonsbeholdning = garb.pensjonsbeholdning;
        delingstall67 = garb.delingstall67;
        sats = garb.sats;
        garPN_tt_anv = garb.garPN_tt_anv;
        garPN_justert = garb.garPN_justert;
        if (garb.satsType != null) {
            satsType = new GarantipenNivaCti(garb.satsType);
        }
        if (garb.justertGarantipensjonsniva != null) {
            justertGarantipensjonsniva = new JustertGarantipensjonsniva(garb.justertGarantipensjonsniva);
        }
    }

    public JustertGarantipensjonsniva getJustertGarantipensjonsniva() {
        return justertGarantipensjonsniva;
    }

    public void setJustertGarantipensjonsniva(JustertGarantipensjonsniva justertGarantipensjonsniva) {
        this.justertGarantipensjonsniva = justertGarantipensjonsniva;
    }

    public double getDelingstall67() {
        return delingstall67;
    }

    public void setDelingstall67(double delingstall67) {
        this.delingstall67 = delingstall67;
    }

    public double getPensjonsbeholdning() {
        return pensjonsbeholdning;
    }

    public void setPensjonsbeholdning(double pensjonsbeholdning) {
        this.pensjonsbeholdning = pensjonsbeholdning;
    }

    public GarantipenNivaCti getSatsType() {
        return satsType;
    }

    public void setSatsType(GarantipenNivaCti satsType) {
        this.satsType = satsType;
    }

    public double getSats() {
        return sats;
    }

    public void setSats(double sats) {
        this.sats = sats;
    }

    public double getGarPN_tt_anv() {
        return garPN_tt_anv;
    }

    public void setGarPN_tt_anv(double garPN_tt_anv) {
        this.garPN_tt_anv = garPN_tt_anv;
    }

    public double getGarPN_justert() {
        return garPN_justert;
    }

    public void setGarPN_justert(double garPN_justert) {
        this.garPN_justert = garPN_justert;
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/Garantipensjonsbeholdning.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.beregning2011.JustertGarantipensjonsniva
import no.nav.pensjon.regler.domain.enum.BeholdningtypeEnum
import no.nav.pensjon.regler.domain.enum.GarantiPensjonsnivaSatsEnum

class Garantipensjonsbeholdning() : Beholdning() {
    var justertGarantipensjonsniva: JustertGarantipensjonsniva? = null
    var pensjonsbeholdning = 0.0
    var delingstallVedNormertPensjonsalder = 0.0

    /**
     * Satstype brukt i garantipensjonsnivå.
     */
    var satsTypeEnum: GarantiPensjonsnivaSatsEnum? = null

    /**
     * Garantipensjonsnivå sats
     */
    var sats = 0.0

    /**
     * Har ektefelle inntekt over 2G ved virk.
     * Bidrar til å avgjøre [satsTypeEnum]
     */
    var ektefelleInntektOver2G: Boolean = false

    /**
     * Garantipensjonsnivå justert for trygdetid
     */
    var garPN_tt_anv = 0.0

    /**
     * Garantipensjonsnivå fremskrevet.
     */
    var garPN_justert = 0.0

    override var beholdningsTypeEnum: BeholdningtypeEnum = BeholdningtypeEnum.GAR_PEN_B
}